include/fbgemm/Utils.h

```diff
@@ -350,4 +354,220 @@ FBGEMM_API void fbgemmPartition1DBlocked(
     int block_size,
     std::int64_t& start,
     std::int64_t& end);
+
+namespace {
+
+// implementation taken from pytorch/c10/util/llvmMathExtras.h
+template <typename T>
+size_t count_leading_zeros(T val) {
+    if (!val)
+      return std::numeric_limits<T>::digits;
+
+    size_t zero_bits = 0;
+    for (T shift = std::numeric_limits<T>::digits >> 1; shift; shift >>= 1) {
+      T tmp = val >> shift;
+      if (tmp)
+        val = tmp;
+      else
+        zero_bits |= shift;
+    }
+    return zero_bits;
+}
+
+// histogram size per thread
+constexpr int RDX_HIST_SIZE = 256;
+
+void combine_prefix_sum(
```

Comment:
There are some redundant codes in `combine_prefix_sum` and `combine_prefix_sum_for_msb`, can we abstract into a function?

Reply:
I just extracted common code to a macro function. PTAL.

Reply:
LGTM.